lib/redix_cluster/options.ex

```diff
@@ -0,0 +1,50 @@
+defmodule RedixCluster.Options do
+  @moduledoc false
+
+  definition = [
+    name: [
+      required: true,
+      type: :atom,
+      doc: "Used for cluster process registration"
+    ],
+    nodes: [
+      required: true,
+      type: {:custom, __MODULE__, :validate_nodes, []},
+      doc: "A list of cluster node URLs"
+    ]
+  ]
+
+  @definition NimbleOptions.new!(definition)
+
+  @valid_schemes ["redis", "rediss"]
+
+  def definition, do: @definition
+
+  def validate_nodes(nodes) when is_list(nodes) do
+    Enum.each(nodes, &parse_node_url/1)
+
+    {:ok, nodes}
+  rescue
+    ex in ArgumentError ->
+      {:error, ex.message}
+  end
+
+  def validate_nodes(nodes) do
+    {:error, "expected :nodes to be list, got: #{inspect(nodes)}"}
+  end
+
+  defp parse_node_url(node_url) when is_binary(node_url) do
```

Comment:
Should we add a trailing bang `parse_node_url!` ?

Reply:
I'll simplify this to return a tuple instead. Also, it's better to return all invalid node_urls instead of the first one.